app/server/__init__.py

```diff
@@ -119,11 +121,20 @@ def before_request():
 
     @app.after_request
     def after_request(response):
-            # Execute any async celery tasks
+        from server.utils import pusher
 
         if response.status_code < 300 and response.status_code >= 200:
             db.session.commit()
 
+        for transaction, queue in g.pending_transactions:
+            transaction.send_blockchain_payload_to_worker(queue=queue)
+
+        for exchange_args in g.pending_exchanges:
+            bt.make_liquid_token_exchange(**exchange_args)
```

Comment:
I'm fully aware that is one is my fault, but I figure now is as good as time as any to fix it lol...

credit transfers and exchanges use a different level of abstractions to send tasks to the blockchain worker:
 credit transfers have a nice `send_blockchain_payload_to_worker` method, where as `pending_exchanges` just has this gnarly bunch of blockchain tasks tied directly into `exchange_from_amount`.

Only really noticed because these two for loops next to each other are totally different. Thoughts on creating a  `send_blockchain_payload_to_worker(queue)` method in Exchange that holds all the blockchain tasks that need to be fired? As an added bonus, it means we could just append all the tasks to `g.pending_transactions` and fire them in a single neat loop.

Reply:
@enjeyw I think that makes sense. I committed this change for you to have a look at. I think there's a (small) compromise, just in that we have to filter out exchanges in the `after_request` pusher batch.